types/user.ts

```diff
@@ -575,6 +576,36 @@ export class User extends LazyObject {
     return updatePassword(auth.currentUser!, newPassword);
   }
 
+  public async changeEmail(
+    oldEmail: string,
+    newEmail: string,
+    password: string
+  ) {
+    await this.checkIfSignedIn();
+    await this.getData();
+    await reauthenticateWithCredential(
+      auth.currentUser!,
+      EmailAuthProvider.credential(this.email!, password)
+    );
+    if (this.email! !== oldEmail)
+      return Promise.reject('Must provide the correct old email!');
+    if (this.status! === UserStatus.Manager)
+      return Promise.reject(
+        'Unable to delete manager account: Please ensure there are other manager accounts and demote your own before deleting.'
+      );
+    if (isKnightsEmail(oldEmail))
+      return Promise.reject(
+        "Unable to change an account's email which is using a knights email."
+      );
+
+    return updateEmail(auth.currentUser!, newEmail).then(() => {
```

Comment:
I'm seeing some people say (with barely any upvotes) that updateEmail() throws an error if the attempted new email already exists on an auth account, but the Firebase docs don't say anything clear about that.
I'm making a comment instead of blocking to see if you know about this